code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm

```diff
@@ -299,21 +299,152 @@
+
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/attacked_by(obj/item/I, mob/living/user)
+	if(!check_edible(I))
+		return ..()
+	eat_atom(I)
+
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/AttackingTarget(atom/attacked_target)
+	if(!check_edible(attacked_target))
+		return ..()
+	eat_atom(attacked_target)
+
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/UnarmedAttack(atom/A)
+	if(!check_edible(A))
+		return ..()
+	eat_atom(A)
+
+///Returns whether or not the supplied movable atom is edible.
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/proc/check_edible(atom/movable/potential_food)
+	if(isliving(potential_food))
+		var/mob/living/living_morsel = potential_food
+		if(living_morsel.mob_size > MOB_SIZE_SMALL)
+			return FALSE
+		else
+			return TRUE
+
+	if(IS_EDIBLE(potential_food))
+		if(prank_pouch.len >= 8)
+			to_chat(src, "<span class='warning'>Your prank pouch is filled to the brim! You don't think you can swallow any more morsels right now.</span>")
+			return FALSE
+		return TRUE
+
+///This proc eats the atom, certain funny items are stored directly in the prank pouch while bananas grant a heal based on their potency and the peels are retained in the pouch.
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/proc/eat_atom(atom/movable/eaten_atom)
+	if(istype(eaten_atom, /obj/item/food/pie/cream) || istype(eaten_atom, /obj/item/food/grown/tomato) || istype(eaten_atom, /obj/item/food/meatclown))
+		eaten_atom.forceMove(src)
+		prank_pouch += eaten_atom
+
+	else if(istype(eaten_atom, /obj/item/food/grown/banana))
+		var/obj/item/food/grown/banana/banana_morsel = eaten_atom
+		adjustBruteLoss(-banana_morsel.seed.potency * 0.25)
+		prank_pouch += banana_morsel.generate_trash(src)
+		qdel(eaten_atom)
+	else
+		qdel(eaten_atom)
+	src.visible_message("<span class='warning>[src] eats [eaten_atom]!</span>", "<span class='notice'>You eat [eaten_atom].</span>")
+	playsound(loc,'sound/items/eatfood.ogg', rand(30,50), TRUE)
+	flick("glutton_mouth", src)
+
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/tamed()
+	. = ..()
+	can_buckle = TRUE
+	buckle_lying = 0
+	var/datum/component/riding/riding = LoadComponent(/datum/component/riding)
+	riding.set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0, 24), TEXT_SOUTH = list(0, 24), TEXT_EAST = list(-16, 24), TEXT_WEST = list(16, 24)))
+	riding.set_vehicle_dir_layer(SOUTH, ABOVE_MOB_LAYER)
+	riding.set_vehicle_dir_layer(NORTH, OBJ_LAYER)
+	riding.set_vehicle_dir_layer(EAST, ABOVE_MOB_LAYER)
+	riding.set_vehicle_dir_layer(WEST, ABOVE_MOB_LAYER)
+
+	//these 0 offsets are needed for the riding component to work properly with base pixel offset mobs for some cursed reason.
+	riding.set_vehicle_dir_offsets(SOUTH, 0, 0)
+	riding.set_vehicle_dir_offsets(NORTH, 0, 0)
+	riding.set_vehicle_dir_offsets(EAST, 0, 0)
+	riding.set_vehicle_dir_offsets(WEST, 0, 0)
+
+	riding.drive_verb = "ride"
```

Comment:
your code does not compile, please resolve this

Reply:
I did